### PR TITLE
Refactors machine_stat and is_processing() to process on demand

### PR DESCRIPTION
**Pull request by Rohesie**
Handles changes in the variable through `set_machine_stat()` so we can react to the event of it changing.
Offers no functional changes here as is, but should allow things that depend on it to process on-demand.
Done for a change request on #53020 

Edit: Went ahead and did a larger refactor. Turned `is_processing()` into a boolean we can keep track the changes of, so we can start and stop processing as it changes.
As a result of this the cyborg recharging station, recharging defibrillator mount, cryo cell and chem dispenser all start and stop processing depending on if they are operational, instead of always processing regardless.

### CODE REVIEW

File: code/modules/reagents/chemistry/machinery/smoke_machine.dm [hunk 1]
```diff
@@ -63,10 +67,16 @@
 		max_range += M.rating
 	max_range = max(3, max_range)
 
+
+/obj/machinery/smoke_machine/on_set_is_operational(old_value)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/machinery/smoke_machine.dm:71]: Might wanna check if there's any machines that process when not operational, I can imagine a few of them migth have some special effects or some other behavior

Comment (Rohesie) [reply code/modules/reagents/chemistry/machinery/smoke_machine.dm:71]: This is redefined only for a few machines that do not process when not operational.
The base proc doesn't do anynthing.

**Comment by LemonInTheDark** [comment]
The use of ?. bothers me, but that's totally style, and not something I should put on you.
Aside from that, this looks good to me, it's tested I trust?

**Comment by Rohesie** [comment]
Yep, the chemistry dispenser stopped and started processing as expected in my local test.

**Comment by LemonInTheDark** [comment]
Wonderful

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -179,6 +178,25 @@ Class Procs:
...
+		return
+	. = machine_stat
+	machine_stat = new_value
+	on_set_machine_stat(.)
+
+
+///Called when the value of `machine_stat` changes, so we can react to it.
+/obj/machinery/proc/on_set_machine_stat(old_value)
+	if(old_value & (NOPOWER|BROKEN|MAINT))
+		if(!(machine_stat & (NOPOWER|BROKEN|MAINT))) //From off to on.
```
Comment (Dennok) [inline code/game/machinery/_machinery.dm:194]: What if make `(NOPOWER|BROKEN|MAINT)` a variable to be able make machinery that continue work when unpowered or on maint? With no need to override this proc. Or blocking of aquirring  NOPOWER flag by powerless machinery.

Comment (Rohesie) [reply code/game/machinery/_machinery.dm:194]: A `machine_operational` variable would be good to have, along with a `set_machine_operational()` proc.
If you want to do that, it'd be an improvement.

**Pull request merged**